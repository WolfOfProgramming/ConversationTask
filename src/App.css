*,
*::after,
*::before {
    padding: 0;
    margin: 0;
    box-sizing: border-box;
}

.chat-container {
    width: 100%;
    height: 100vh;
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: #e4e4e4;
}

.chat-window {
    width: 100%;
    max-width: 500px;
    height: 100%;
    max-height: 800px;
    background-color: #ffffff;
    display: flex;
    flex-direction: column;
    overflow: hidden;
}

.chat-window__switch {
    display: flex;
    flex-grow: 3;
}

.chat-window__button {
    width: 50%;
    display: flex;
    justify-content: center;
    align-items: center;
    border: none;
    background-color: #ffffff;
    cursor: pointer;
}

.chat-window__button:hover {
    background-color: #f7f7f7;
    border-bottom: 2em solid #3a6187;
}

.chat-window__button-text {
    font-size: 2em;
}

.chat-window__description {
    font-size: 2em;
    padding: 1.25em;
    background-color: #747474;
    color: #ffffff;
    text-align: center;
}

.chat-window__container {
    display: flex;
    justify-content: space-between;
    background-color: #e4e4e4;
    height: 15%;
}

.chat-window__controls {
    display: flex;
    padding: 1em 0;
    align-items: center;
    justify-content: space-between;
    color: #3a6187;
    width: 30%;
}

.chat-window__logout {
    border: none;
    background-color: transparent;
    color: #3a6187;
    font-size: 1em;
    cursor: pointer;
}

.chat-window__user-name {
    padding: 0.5em;
}

.chat-window__dialog {
    display: flex;
    flex-direction: column;
    height: 85%;
}

.chat-window__chat-person {
    text-align: center;
    padding: 1.2em;
    font-size: 1.125em;
    color: #747474;
    border-bottom: 2px solid #f7f7f7;
    flex-shrink: 0;
}

.chat-window__dialog-window {
    list-style: none;
    display: flex;
    flex-direction: column;
    height: 55%;
    flex-shrink: 0;
    overflow: auto;
    padding: 1em;
    flex-grow: 1;
}

.chat-window__message {
    padding: 0.75em;
    margin: 0.5em;
    align-self: flex-start;
    background-color: #eeeeee;
    color: #747474;
    border-radius: 1em;
    max-width: 60%;
}

.chat-window__message--active {
    background-color: #7298be;
    color: #ffffff;
    align-self: flex-end;
}

.chat-window__dialog-form {
    height: 100px;
    display: flex;
    align-content: center;
    border-top: 1em solid #3a6187;
    flex-shrink: 0;
    padding: 1em;
    background-color: #eeeeee;
    justify-content: space-around;
}

.chat-window__textarea {
    height: 100%;
    resize: none;
    border: none;
    border-radius: 1em;
    width: 70%;
    padding: 0.5em;
}

.chat-window__submit {
    border: none;
    background-color: transparent;
    color: #3a6187;
    font-size: 1.25em;
    cursor: pointer;
}

.chat-window__name {
    color: #3a6187;
    text-align: center;
    background-color: #e4e4e4;
    padding: 1em 0;
}

.chat-window__title {
    font-size: 2.375em;
}

.chat-window__title--small {
    font-size: 0.875em;
}
