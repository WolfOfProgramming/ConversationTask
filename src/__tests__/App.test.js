import React from 'react';
import App from '../App';
import renderer from 'react-test-renderer';

it('renders standart app structure', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
});
