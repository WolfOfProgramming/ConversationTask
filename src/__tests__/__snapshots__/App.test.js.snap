// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders everything properly 1`] = `
<div
  className="chat-container"
>
  <div
    className="chat-window"
  >
    <div
      className="chat-window__name"
    >
      <p
        className="chat-window__title"
      >
        Chat
      </p>
      <p
        className="chat-window__title chat-window__title--small"
      >
        To JS Master
      </p>
    </div>
    <div
      className="chat-window__switch"
    >
      <button
        className="chat-window__button"
        onClick={[Function]}
      >
        <span
          className="chat-window__button-text"
        >
          John Smith
        </span>
      </button>
      <button
        className="chat-window__button"
        onClick={[Function]}
      >
        <span
          className="chat-window__button-text"
        >
          John Doe
        </span>
      </button>
    </div>
    <p
      className="chat-window__description"
    >
      Choose your account and start a conversation
    </p>
  </div>
</div>
`;

exports[`renders standart app structure 1`] = `
<div
  className="chat-container"
>
  <div
    className="chat-window"
  >
    <div
      className="chat-window__name"
    >
      <p
        className="chat-window__title"
      >
        Chat
      </p>
      <p
        className="chat-window__title chat-window__title--small"
      >
        To JS Master
      </p>
    </div>
    <div
      className="chat-window__switch"
    >
      <button
        className="chat-window__button"
        onClick={[Function]}
      >
        <span
          className="chat-window__button-text"
        >
          John Smith
        </span>
      </button>
      <button
        className="chat-window__button"
        onClick={[Function]}
      >
        <span
          className="chat-window__button-text"
        >
          John Doe
        </span>
      </button>
    </div>
    <p
      className="chat-window__description"
    >
      Choose your account and start a conversation
    </p>
  </div>
</div>
`;
